src/Chapter2.hs

```diff
@@ -841,7 +851,13 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n l | n < 0 = []
+           | otherwise = drop r $ take n' (cycle l) 
```

Comment:
Unfortunately, `cycle` fails in runtime on empty lists :recycle: So you need to handle this case separately for this function to work properly.